src/webpack/index.ts

```diff
@@ -75,3 +84,55 @@ function removeMiniCssExtract(config: any) {
     }
   });
 }
+
+function getLibraryName(options: any) {
+  if (options?.customWebpackConfig?.libraryName) {
+    return options.customWebpackConfig.libraryName;
+  }
+
+  const projectName = getProjectNameFromAngularJson();
+  if (projectName) return projectName;
+
+  console.warn(
+    'Warning: single-spa-angular could not determine a library name to use and has used a default value.',
+  );
+  console.info('This may cause issues if this app uses code-splitting or lazy loading.');
+  if (!options) {
+    console.info('You may also need to update extra-webpack.config.json.');
+  }
+  console.info('See <SomeUrl> for information on how to resolve this.');
+
+  return 'angular_single_spa_project';
+}
+
+function getProjectNameFromAngularJson(): string | null {
+  const angularJsonPath = findUp(['angular.json', '.angular.json']);
+  if (!angularJsonPath) return null;
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const angularJson = require(angularJsonPath);
+  if (!angularJson.projects) return null;
+
+  var projects = Object.getOwnPropertyNames(angularJson.projects);
+
+  if (projects.length === 1) return projects[0];
+
+  // If we reach here there are multiple (or zero) projects in angular.json
+  // we cannot tell which one to use, so we will end up using the default.
+  return null;
+}
+
+function findUp(files: string[]): string | null {
+  let current = process.cwd();
+  const root = path.parse(current).root;
+
+  while (current !== root) {
```

Comment:
This looks like an infinite loop - we are never updating `current` inside of the loop.

Reply:
Yeah, the while loop is broken. I must have accidentally lost a line.

Reply:
Fixed in latest push.